server/channels/store/sqlstore/channel_store.go

```diff
@@ -1547,16 +1547,14 @@ func (s SqlChannelStore) getByName(teamId string, name string, includeDeleted bo
 	query := s.getQueryBuilder().
 		Select("*").
 		From("Channels").
-		Where(sq.Eq{"Name": name})
-
-	if !includeDeleted {
-		query = query.Where(sq.Eq{"DeleteAt": 0})
-	}
-	if teamId != "" {
-		query = query.Where(sq.Or{
+		Where(sq.Eq{"Name": name}).
+		Where(sq.Or{
 			sq.Eq{"TeamId": teamId},
 			sq.Eq{"TeamId": ""},
 		})
```

Comment:
The original [POC](https://github.com/mattermost/mattermost/commit/aa59c28b04fa45f0b02e5149ac0fbcd2b78420b9) changed the semantics here to treat `teamId != ""` as allowing any team, but in at least one case we assumed `teamId == ""` meant only direct channels:

https://github.com/mattermost/mattermost/blob/823ef27d6c5351925de667da243df4bf458ce07b/server/channels/app/channel.go#L3491-L3492

To keep things consistent without a clear reason to keep the wider search, I reverted back to the old semantics for `C.TeamId = :teamId OR c.TeamId = ""` everywhere, while preserving the SQL Builder changes where applicable.